pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -514,6 +539,21 @@ func subnetsBelongToSameVPC(subnets []*ec2.Subnet) bool {
 	return len(m) == 1
 }
 
+// getClusterVpcID returns the cluster vpc id from cluster infra stack.
+func (c Cloud) getClusterVpcID(name string) (string, error) {
+	s, err := c.getStack(makeClusterInfraStackName(name))
```

Comment:
(as discussed offline) May be worth wrapping this string in a new type that is then passed around, to avoid functions having to "manually" do the `makeClusterInfraStackName` mapping (e.g. `clusterStackName.value` would return the mapped value).